initial commit of getting started migration (#5)
AdobeDocs/commerce-webapi: src/pages/get-started/create-integration.md

<th>Element</th>
<th>Description</th>
</tr>
<tr>
<td>integrations</td>
<td>Contains one or more integration definitions.</td>
</tr>
<tr>
<td>integration name=""</td>
<td>Defines an integration. The <inlineCode class="spectrum-Body--sizeS">name</inlineCode> must be specified.</td>
</tr>
<tr>
<td>email</td>
<td>An email to associate with this integration.</td>
</tr>
<tr>
<td>endpoint_url</td>
<td><p>Optional. The [URL](https://glossary.magento.com/url) where OAuth credentials can be sent when using OAuth for token exchange. We strongly recommend using <inlineCode class="spectrum-Body--sizeS">https://</inlineCode>.</p>
<p>See <a href="./authentication/gs-authentication-oauth">OAuth-based authentication</a> for details.</p></td>
</tr>
<tr>
<td>identity_link_url</td>
<td>Optional. The URL that redirects the user to link their 3rd party account with the Commerce integration.</td>
</tr>
</table>

## Install your module

Use the following steps to install your module:

1. Run the following command to update the [database schema](https://glossary.magento.com/database-schema) and data.

   ```bash
   bin/magento setup:upgrade
   ```

1. Run the following command to generate the new code.

>     <InlineAlert variant="info" slots="text"/>

Reviewer: This is another multiparagraph note that doesn't render. I would create an unofficial note like this: **Note:** In Production mode...

Author: it seems we can use `<div>` tags to include multiple paragraphs.

>    In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. Magento does not prompt you to run the compile command in Developer mode.

Reviewer: ```suggestion
   In Production mode, you may receive a message to 'Please rerun Magento compile command'.  Enter the command below. You are not prompted to run the compile command in Developer mode.
```

   ```bash
   bin/magento setup:di:compile
   ```

1. Run the following command to clean the cache.

   ```bash
   bin/magento cache:clean
   ```

## Check your integration

Log in to Magento and navigate to **System > Extensions > Integrations**. The integration should be displayed in the grid.

## Integrate with your application

Before you can activate your integration in Magento, you must create two pages on your application to handle OAuth communications.

*  The location specified in the `identity_link_url` parameter must point to a page that can handle login requests.

*  The location specified in the `endpoint_url` parameter (**Callback URL** in Admin) must be able to process OAuth token exchanges.

### Login page

When a merchant clicks the **Activate** button in Admin, a pop-up login page for the third-party application displays. Magento sends values for `oauth_consumer_key` and `success_call_back` parameters. The application must store the value for `oauth_consumer_key` to tie it to the login ID. Use the `success_call_back` parameter to return control back to Magento.

### Callback page

The callback page must be able to perform the following tasks:

> *  Receive an initial HTTPS POST that Magento sends when the merchant activates integration. This post contains the Magento store URL, an `oauth_verifier`, the OAuth consumer key, and the OAuth consumer secret. The consumer key and secret are generated when the integration is created.

Reviewer: ```suggestion
*  Receive an initial HTTPS POST that Commerce sends when the merchant activates integration. This post contains the Commerce store URL, an `oauth_verifier`, the OAuth consumer key, and the OAuth consumer secret. The consumer key and secret are generated when the integration is created.
```

*  Ask for a request token. A request token is a temporary token that the user exchanges for an access token. Use the following API to get a request token from Magento:

   `POST /oauth/token/request`

   See [Get a request token](./authentication/gs-authentication-oauth.md#get-a-request-token) for more details about this call.

*  Parse the request token response. The response contains an `oauth_token` and `oauth_token_secret`.

> *  Ask for an access token. The request token must be exchanged for an access token. Use the following API to get a request token from Magento:

Reviewer: ```suggestion
*  Ask for an access token. The request token must be exchanged for an access token. Use the following API to get a request token:
```

   `POST /oauth/token/access`

   See [Get an access token](./authentication/gs-authentication-oauth.md#get-an-access-token/) for more details about this call.

*  Parse the access token response. The response contains an `oauth_token` and `oauth_token_secret`. These values will be different than those provided in the request token response.

> *  Save the access token and other OAuth parameters. The access token and OAuth parameters must be specified in the `Authorization` header in each call to Magento.

Reviewer: ```suggestion
*  Save the access token and other OAuth parameters. The access token and OAuth parameters must be specified in the `Authorization` header in each call to Commerce.
```

## Related Topics

*  [Web API authentication](./authentication/)
*  [OAuth-based authentication](./authentication/gs-authentication-oauth.md)

> *  [Magento System Requirements](https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements.html)

Reviewer: ```suggestion
*  [System Requirements](https://devdocs.magento.com/guides/v2.4/install-gde/system-requirements.html)
```
